script.js

```diff
@@ -1,4 +1,95 @@
+var diceResult;
+var mode = "Player 1 Roll";
+var roll1;
+var roll2;
+var player1Result;
+var player2Result;
+var compareResult;
+var player1Score = 0;
+var player2Score = 0;
+var player1Record = [];
+var player2Record = [];
+
+var rollDice = function () {
+  diceResult = Math.floor(Math.random() * 6) + 1;
+  return diceResult;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  console.log("Player 1 and 2 scores: ", player1Score, player2Score);
+
+  if (mode == "Player 1 Roll") {
+    roll1 = rollDice();
+    roll2 = rollDice();
+    console.log("Player 1 rolls are: ", roll1, roll2);
+    var myOutputValue = `Welcome Player 1. <br>Die 1 is ${roll1} and Die 2 is ${roll2}.<br>
+  Please enter 1 or 2 to choose which number goes first.`;
+    mode = "Player 1 Choose";
+    return myOutputValue;
+  }
+
+  if (mode == "Player 1 Choose") {
+    if (input == 1) {
+      player1Result = roll1 * 10 + roll2;
+    }
+    if (input == 2) {
+      player1Result = roll2 * 10 + roll1;
+    }
+    player1Record.push(player1Result);
+    console.log("Player 1 generated: ", player1Result);
+    var myOutputValue = `Player 1's generated number is ${player1Result}.<br>
+    It is Player 2's turn to roll the dice. Please press the button.`;
+    mode = "Player 2 Roll";
+    return myOutputValue;
+  }
+
+  if (mode == "Player 2 Roll") {
+    roll1 = rollDice();
+    roll2 = rollDice();
+    console.log("Player 2 rolls are: ", roll1, roll2);
+    var myOutputValue = `Welcome Player 2. <br>Die 1 is ${roll1} and Die 2 is ${roll2}.<br>
+  Please enter 1 or 2 to choose which number goes first.`;
+    mode = "Player 2 Choose";
```

Comment:
[Suggestion] maybe can consider having helper functions outside of main. With helper functions you can use the same function for both player 1 and player 2

some like
var playerRoll = (player) {
}

When it is player 1 you can run playerRoll(player1)